File: website/blueprints/auction/models/Auction.py
from flask_sqlalchemy import SQLAlchemy
from datetime import datetime

db = SQLAlchemy()

class Auction(db.Model):
    __tablename__ = "auctions"

    id = db.Column(db.Integer, primary_key=True)
    item_id = db.Column(db.String(255), nullable=False)  # itemID in Mongoose
    item_name = db.Column(db.String(255), nullable=False)  # itemName in Mongoose
    seller_id = db.Column(db.String(255), nullable=False)  # sellerID in Mongoose
    starting_price = db.Column(db.Float, nullable=False)  # startingPrice in Mongoose
    current_bid = db.Column(db.Float, default=0.0)  # currentBid in Mongoose
    current_bidder_id = db.Column(db.String(255))  # currentBidderID in Mongoose
    end_time = db.Column(db.DateTime, nullable=False)  # endTime in Mongoose
    status = db.Column(db.String(50), default="active", nullable=False)  # status in Mongoose

    def __repr__(self):
        return f"<Auction {self.item_name} - {self.status}>"

File: website/blueprints/auction/models/Auction.js
const mongoose = require("mongoose");

const AuctionSchema = new mongoose.Schema({
  itemID: { type: String, required: true },
  itemName: { type: String, required: true },
  sellerID: { type: String, required: true },
  startingPrice: { type: Number, required: true },
  currentBid: { type: Number, default: 0 },
  currentBidderID: { type: String },
  endTime: { type: Date, required: true },
  status: { type: String, enum: ["active", "completed"], default: "active" },
});

module.exports = mongoose.model("Auction", AuctionSchema);
